Hyperion/SerializerFactories/EnumerableSerializerFactory.cs

```diff
@@ -68,33 +75,84 @@ private static Type GetEnumerableType(Type type)
                 .FirstOrDefault();
         }
 
+        private static ConstructorInfo GetEnumerableConstructor(Type type)
+        {
+            var enumerableType = GetEnumerableType(type);
+            return enumerableType != null
+              ? type.GetTypeInfo().GetConstructor(new[] { typeof(IEnumerable<>).MakeGenericType(enumerableType) })
+              : null;
+        }
+
+        private static Func<object, object> CompileCtorToDelegate(ConstructorInfo ctor, Type argType)
```

Comment:
We need to check which is faster: using lambda `Func<,>` which call the constructor, or calling `ConstructorInfo.Invoke` directly and use that one.

Reply:
+1 Is there any standards for benchmarking I should follow, e.g. which benchmarking library to use, what data to test on, etc.?